nexus_constructor/instrument_view.py

```diff
@@ -226,8 +255,8 @@ def delete_component(self, name):
         Delete a component from the InstrumentView by removing the components and entity from the dictionaries.
         :param name: The name of the component.
         """
-
-        self.component_entities[name].setParent(None)
+        for entity in self.component_entities[name]:
+            entity.setParent(None)
```

Comment:
shouldn't this be done with the context manager anyway? could it be done with `del` instead?

Reply:
The context manager is to temporarily disconnect the entities. In this case we never want to reconnect them.
`del` does not do the same thing, if we delete the component before disconnecting it then we get segfaults.